### PR TITLE
Stutter effect now easily translateable to other languages

### PR DESCRIPTION
**Pull request by TokenStyle**
## About the PR
Move sttuter regex from StutteringSystem.cs into .ftl file, to easier translate stutter effect

## Technical details
Easier translate/change regex of .ftl neither code changing.
Couldn't Stutter variable be placed in class as field, because Initialized or constructor are not an option, because method Accentuate could start without Initialization, and contrustor could be started before Loc even inialized.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
For other projects, should change their locales like: Resources/Locale/en-US/accent/stutter.ftl , in order to change regex. As it was, these changes are coded instead translation.

### CODE REVIEW

File: Content.Server/Speech/EntitySystems/StutteringSystem.cs [hunk 1]
```diff
@@ -37,15 +33,16 @@ private void OnAccent(EntityUid uid, StutteringAccentComponent component, Accent
         public string Accentuate(string message, StutteringAccentComponent component)
         {
             var length = message.Length;
-
             var finalMessage = new StringBuilder();
-
             string newLetter;
 
+            // get regex accent according to stutter.ftl
+            Regex stutter = new(Loc.GetString("stutter-speech-regex-accent"), RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
Comment (deltanedas) [inline Content.Server/Speech/EntitySystems/StutteringSystem.cs:40]: this will recompile the regex for every message

Comment (TokenStyle) [reply Content.Server/Speech/EntitySystems/StutteringSystem.cs]: Yeah fixed

**Comment by deltanedas** [comment]
its still bad
cache the Regex instance with a dictionary or something

**Comment by TokenStyle** [comment]
Though, a lot of places where Regex'es are hardcoded as:
`public static readonly Regex TagRegex = new("^[a-zA-Z0-9, ]*$", RegexOptions.Compiled);`
Be better to move it abstract like `class RegexLoc : Regex` and put it on RobustToolkit engine as module that let's you use Compiled regex with current architecture in `Initialize` or other places instead of putting in on Constructor (Loc.GetString then will not work) or other methods (RegexOptions.Compiled is not an option because could be executed multiple times so regex's not fast).

Though, strings in current state are not bottleneck or somethin', and it's not that important

**Comment by TokenStyle** [comment]
Looks like a lot of work, if someone would like they could make that module and other contributers could refactor all hardcoded regexes into translateable one

**Comment by TokenStyle** [comment]
Feel free to continue that PR, or refer it on [RobustToolkit](https://github.com/space-wizards/RobustToolbox/tree/master) PR if someone would like to create a module

**Comment by slarticodefast** [comment]
Since it seems like you are no longer working on this I will be closing this PR for now. If you come back to it, leave a comment and I will re-open it.

>Looks like a lot of work, if someone would like they could make that module and other contributers could refactor all hardcoded regexes into translateable one

That sounds like a good suggestion (I'm no expert on regex though). You should open an issue for it.

**Pull request closed without merging**